time_stamper_cv/src/image_processor.cpp

```diff
@@ -110,3 +118,17 @@ void ImageProcessor::visualizeCorners(const cv::Mat &visualization_mat, PointAng
 void ImageProcessor::log(const std::string &message) {
   ROS_INFO("%s", message.c_str());
 }
+
+void ImageProcessor::fillInLedStateMessage(time_stamper_cv::Ledstate* msg) {
+  msg->counter = led_parser_->getBinaryValue(ImageProcessorConfig::BOTTOM_ROW);
+  msg->is_valid = convex_shape_->isShapeValid();
+
+  std::vector<std::string> rows = cfg_.rows();
+  std::for_each(rows.begin(), rows.end(), [this, &msg] (const std::string& name) {
+    for (int i = 0; i < led_parser_->getLedRow(name).size(); i++) {
```

Comment:
This lookup `led_parser_->getLedRow(name)` indicates to me that actually, you should directly iterate over `led_rows` map's items (ideally with c++17 structured binding) to avoid the ugly `.first` and `.second`..